ciao-controller/external_ip.go

```diff
@@ -180,10 +180,16 @@ func (c *controller) ListMappedAddresses(tenant *string) []types.MappedIP {
 	return IPs
 }
 
-func (c *controller) MapAddress(poolName *string, instanceID string) (err error) {
+func (c *controller) MapAddress(tenantID string, poolName *string, instanceID string) (err error) {
 	var m types.MappedIP
+	var i *types.Instance
 
-	i, err := c.ds.GetInstance(instanceID)
+	if tenantID == "public" {
+		// we allow the admin to map anyone's instance
```

Comment:
in the case of forcibly unmapping an external IP, I think it makes sense. In the case of mapping the external IP for a user, I think it could also make sense but the argument isn't as strong.